RecoTracker/LSTCore/interface/alpaka/Module.h

```diff
@@ -0,0 +1,368 @@
+      return (isInvertedx) ? !(detId & 1) : (detId & 1);
+    };
+
+    static unsigned int parsePartnerModuleId(unsigned int detId, bool isLowerx, bool isInvertedx) {
+      return isLowerx ? (isInvertedx ? detId - 1 : detId + 1) : (isInvertedx ? detId + 1 : detId - 1);
+    };
+
+    template <typename TBuff>
+    void setData(const TBuff& modulesbuf) {
+      detIds = alpaka::getPtrNative(modulesbuf.detIds_buf);
+      moduleMap = alpaka::getPtrNative(modulesbuf.moduleMap_buf);
+      mapdetId = alpaka::getPtrNative(modulesbuf.mapdetId_buf);
+      mapIdx = alpaka::getPtrNative(modulesbuf.mapIdx_buf);
+      nConnectedModules = alpaka::getPtrNative(modulesbuf.nConnectedModules_buf);
+      drdzs = alpaka::getPtrNative(modulesbuf.drdzs_buf);
+      dxdys = alpaka::getPtrNative(modulesbuf.dxdys_buf);
+      nModules = alpaka::getPtrNative(modulesbuf.nModules_buf);
+      nLowerModules = alpaka::getPtrNative(modulesbuf.nLowerModules_buf);
+      partnerModuleIndices = alpaka::getPtrNative(modulesbuf.partnerModuleIndices_buf);
+
+      layers = alpaka::getPtrNative(modulesbuf.layers_buf);
+      rings = alpaka::getPtrNative(modulesbuf.rings_buf);
+      modules = alpaka::getPtrNative(modulesbuf.modules_buf);
+      rods = alpaka::getPtrNative(modulesbuf.rods_buf);
+      subdets = alpaka::getPtrNative(modulesbuf.subdets_buf);
+      sides = alpaka::getPtrNative(modulesbuf.sides_buf);
+      eta = alpaka::getPtrNative(modulesbuf.eta_buf);
+      r = alpaka::getPtrNative(modulesbuf.r_buf);
+      isInverted = alpaka::getPtrNative(modulesbuf.isInverted_buf);
+      isLower = alpaka::getPtrNative(modulesbuf.isLower_buf);
+      isAnchor = alpaka::getPtrNative(modulesbuf.isAnchor_buf);
+      moduleType = alpaka::getPtrNative(modulesbuf.moduleType_buf);
+      moduleLayerType = alpaka::getPtrNative(modulesbuf.moduleLayerType_buf);
+      sdlLayers = alpaka::getPtrNative(modulesbuf.sdlLayers_buf);
+      connectedPixels = alpaka::getPtrNative(modulesbuf.connectedPixels_buf);
+    }
+  };
+
+  template <typename TDev>
+  struct modulesBuffer : modules {
+    Buf<TDev, unsigned int> detIds_buf;
+    Buf<TDev, uint16_t> moduleMap_buf;
+    Buf<TDev, unsigned int> mapdetId_buf;
+    Buf<TDev, uint16_t> mapIdx_buf;
+    Buf<TDev, uint16_t> nConnectedModules_buf;
+    Buf<TDev, float> drdzs_buf;
+    Buf<TDev, float> dxdys_buf;
+    Buf<TDev, uint16_t> nModules_buf;
+    Buf<TDev, uint16_t> nLowerModules_buf;
+    Buf<TDev, uint16_t> partnerModuleIndices_buf;
+
+    Buf<TDev, short> layers_buf;
+    Buf<TDev, short> rings_buf;
+    Buf<TDev, short> modules_buf;
+    Buf<TDev, short> rods_buf;
+    Buf<TDev, short> subdets_buf;
+    Buf<TDev, short> sides_buf;
+    Buf<TDev, float> eta_buf;
+    Buf<TDev, float> r_buf;
+    Buf<TDev, bool> isInverted_buf;
+    Buf<TDev, bool> isLower_buf;
+    Buf<TDev, bool> isAnchor_buf;
+    Buf<TDev, ModuleType> moduleType_buf;
+    Buf<TDev, ModuleLayerType> moduleLayerType_buf;
+    Buf<TDev, int> sdlLayers_buf;
```

Comment:
Please replace all the buffers with a SoA that uses a single buffer.
The one exception could be `connectedPixels_buf` that has a different size.

Reply:
Is this a comment specific to this ES level data?

Reply:
Probably not, I may leave similar comments as I review more of the code.